v2/slogger/logger_test.go

```diff
@@ -364,3 +365,29 @@ func logHelloMongoDB(logger *Logger) {
 func logHelloWorld(logger *Logger) {
 	logger.logf(WARN, NoErrorCode, "Hello World", nil)
 }
+
+func TestErrorWrapping(t *testing.T) {
+	wrappedErr := ErrorWithCode{
+		ErrCode: NoErrorCode,
+		Err:     customError{},
+	}
+
+	cause := errors.Unwrap(wrappedErr)
+	if cause == nil {
+		t.Error("error cause should not be nil")
+		return
+	}
+	if cause.Error() != "foo" {
+		t.Errorf("cause should be 'foo' but was %s", cause.Error())
+		return
+	}
+	if _, isRightType := cause.(customError); !isRightType {
+		t.Error("error cause is not the right type")
+	}
```

Comment:
Please also test this with errors.Is and errors.As .  See https://go.dev/blog/go1.13-errors

Reply:
changed